android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -345,24 +370,36 @@ private void setSSIDHidden(MethodCall poCall, Result poResult) {
      * string otherwise.
      */
     private void getWiFiAPPreSharedKey(Result poResult) {
-        android.net.wifi.WifiConfiguration oWiFiConfig = moWiFiAPManager.getWifiApConfiguration();
-        if (oWiFiConfig != null && oWiFiConfig.preSharedKey != null) {
-            poResult.success(oWiFiConfig.preSharedKey);
-            return;
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+            android.net.wifi.WifiConfiguration oWiFiConfig = moWiFiAPManager.getWifiApConfiguration();
+
+            if (oWiFiConfig != null && oWiFiConfig.preSharedKey != null) {
+                poResult.success(oWiFiConfig.preSharedKey);
+                return;
+            }
+
+            poResult.error("Exception", "Wifi AP not Supported", null);
         }
-        poResult.error("Exception", "Wifi AP not Supported", null);
+            else {
+                poResult.error("Exception [getWiFIAPPreSharedKey]", "Getting WiFi AP password is not supported on API level >= 26", null);
+            }
```

Comment:
can we retun  "random pre shared key" generated for LocalOnlyHostspot?

Reply:
See SSID comment above.

Reply:
will be tracked in new issue #134